src/languages/yaml.js

```diff
@@ -73,14 +73,10 @@ export default function(hljs) {
         excludeEnd: true,
         relevance: 0
       },
-      { // local tags
+      { // tags
         className: 'type',
         begin: '!' + YAML_TAG_RE,
       },
-      { // data type
-        className: 'type',
-        begin: '!!' + YAML_TAG_RE,
-      },
```

Comment:
I took it out because the spec allows for '!' in tags, and I thought it would be confusing to leave in a pattern that never matches. I can see that leaving in the pattern could be useful, for example, if eventually different highlights were wanted for the two patterns. What if we allow '!' everywhere except the first character (after the !), would that be an appropriate solution?

Reply:
You might be right. I think it's original intent was to note that technically `!!` tags are a different type of tag, but whether that's useful or not I'm not sure.   Generally when writing new grammars I try and keep them as vague as possible why still doing the job well but when maintaining old ones I tend to not remove complexity without completely understanding WHY it was added in the first place.

Reply:
Plus technically I think a lot of what we're talking about are tag prefixes... The tag portion can only contain ASCII, but I don't think we're differentiating.